tutoroars/templates/oars/apps/data/assets/datasets/OpenedX_Clickhouse/xAPI_Events_Course_Graph.yaml

```diff
@@ -0,0 +1,90 @@
+table_name: xAPI Events Course Graph
+main_dttm_col: null
+description: null
+default_endpoint: null
+offset: 0
+cache_timeout: null
+schema: coursegraph
+sql: select cg.display_name, splitByChar('/', xapi_events.course_id)[-1] as course_key,
+  if(event.object.id like '%/xblock/%', splitByChar('/', event.object.id)[5], '')
+  as block_key, event.result.success as success, event.object.definition.extensions.'http://id.tincanapi.com/extension/attempt-id'
+  as attempts from xapi.xapi_events_all_parsed as xapi_events, coursegraph.coursegraph_nodes
+  as cg where cg.location = block_key
```

Comment:
To make this dataset more broadly useful, it also needs to handle course type block_keys, and include all of the xapi data:

```
select cg.display_name, 
splitByChar('/', xapi_events.course_id)[-1] as course_key, 
if(event.object.id like '%/xblock/%', 
    splitByChar('/', event.object.id)[5], 
    format('block-v1:{0}+type@course+block@course', splitByChar(':', course_key)[2])
  ) as block_key, 
event.result.success as success,
event.object.definition.extensions.'http://id.tincanapi.com/extension/attempt-id' as attempts, 
xapi_events.*
FROM xapi.xapi_events_all_parsed as xapi_events, coursegraph.coursegraph_nodes as cg 
where block_key = cg.location
```

Reply:
Yeah I locally had a version of this which outer joined the coursegraph data so that everything would show up, but it's messy. I think that post-conference we should make a plan for the different datasets and document their quirks.

Reply:
Yes -- a plan would be great :)

I also think there's a better place for a lot of this detail, like a Clickhouse view.